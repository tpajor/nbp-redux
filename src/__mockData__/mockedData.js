const card = { id: 'someId', code: 'USD', currency: 'dolar amerykański', rates: [{ ask: 3.3, bid: 3.4 }]};
const card2 = { id: 'someId2', code: 'USD', currency: 'dolar amerykański', rates: { ask: 3.3, bid: 3.4 }};

const search = {
  cards: [card, card2],
  temporaryCard: card,
  searchError: false,
  addCardError: true,
  addCardErrorMessage: 'error',
  currenciesTable: ['usd', 'chf', 'aud'],
  currencyCodeToViewInDetail: 'usd',
};

const user = { userName: 'User', pass: 'pass' };
const user2 = { userName: 'User2', pass: 'pass2'};

export const login = {
  signingError: true,
  signingErrorMessage: 'error',
  signedIn: true,
  userSignedIn: 'User',
  users: [user, user2]
};

const lastRates = [
  { ask: 3.5983, bid: 3.5271, effectiveDate: "2018-05-07", no: "087/C/NBP/2018" },
  { ask: 3.6088, bid: 3.5374, effectiveDate: "2018-05-08", no: "088/C/NBP/2018" }
];

const detail = {
  lastRates: lastRates,
  detailsInputError: false,
  detailsInputErrorMessage: '',
};

export const initialState = { search, login, detail };

export const searchViewData = {
  cards: search.cards,
  temporaryCard: search.temporaryCard,
  searchError: search.searchError,
  signedIn: login.signedIn,
  userSignedIn: login.userSignedIn,
  addCardError: search.addCardError,
  addCardErrorMessage: search.addCardErrorMessage,
  currenciesTable: search.currenciesTable,
};

export const detailViewData = {
  currencyCodeToViewInDetail: search.currencyCodeToViewInDetail,
  lastRates: detail.lastRates,
  detailsInputError: detail.detailsInputError,
  detailsInputErrorMessage: detail.detailsInputErrorMessage,
  signedIn: login.signedIn,
};