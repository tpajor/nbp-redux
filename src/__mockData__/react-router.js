const Link = () => {
  return jest.fn();
}

export default Link;