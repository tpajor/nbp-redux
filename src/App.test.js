import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import AppConnected, {App} from './App';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { initialState } from './__mockData__/mockedData';

Enzyme.configure({ adapter: new Adapter() });

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

const store = mockStore(initialState);

function setup() {
  const props = {
    fetchUsers: jest.fn(),
  };

  const enzymeWrapper = shallow(<App {...props} />);

  return {
    enzymeWrapper,
    props
  };
};

describe('App component', () => {
  it('should render self and subcomponents with their props', () => {
    const { enzymeWrapper } = setup();

    expect(enzymeWrapper.length).toBe(1);

    expect(enzymeWrapper.find('div').length).toBe(1);

    expect(enzymeWrapper.find('Route').length).toBe(2);

    expect(enzymeWrapper.find('Route').first().props().path).toEqual('/');

    expect(enzymeWrapper.find('Route').last().props().path).toEqual('/login');
  });

  it('should render correctly', () => {
    const { enzymeWrapper } = setup();

    expect(enzymeWrapper).toMatchSnapshot();
  });

  it('should map actions to props', () => {
    const enzymeWrapper = shallow(<AppConnected store={store} />);

    expect(typeof enzymeWrapper.props().fetchUsers).toBe('function');
  });

  it('should call fetchUsers on mount', () => {
    const { enzymeWrapper, props } = setup();
    
    expect(props.fetchUsers.mock.calls.length).toBe(1);
    
    const instance = enzymeWrapper.instance();
    instance.componentWillMount();
    expect(props.fetchUsers.mock.calls.length).toBe(2);
  });
});