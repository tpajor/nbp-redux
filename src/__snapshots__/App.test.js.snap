// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    fetchUsers={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <div>
        <Route
                component={[Function]}
                exact={true}
                path="/"
        />
        <Route
                component={[Function]}
                path="/login"
        />
</div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Route
            component={[Function]}
            exact={true}
            path="/"
/>,
          <Route
            component={[Function]}
            path="/login"
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/login",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <div>
          <Route
                    component={[Function]}
                    exact={true}
                    path="/"
          />
          <Route
                    component={[Function]}
                    path="/login"
          />
</div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Route
              component={[Function]}
              exact={true}
              path="/"
/>,
            <Route
              component={[Function]}
              path="/login"
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/login",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
